extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -333,13 +333,28 @@ fn generate_markdown_from_item_data(
     if let Some(sig) = &doc_item.signature() {
         if !sig.is_empty() {
             // TODO(#1525) add cairo support to mdbook
-            writeln!(&mut markdown, "```rust\n{sig}\n```\n")?;
+            let signature_formatted = format!(
+                "<pre><code class=\"language-rust\">{}</code></pre>\n",
+                escape_html_markers(sig)
+            );
+            writeln!(&mut markdown, "{}", signature_formatted)?;
         }
     }
-
     Ok(markdown)
 }
 
+fn escape_html_markers(input: &str) -> String {
```

Comment:
I wonder if we might need to escape more characters than this? (like & ' ") Can you elaborate why yes or why not? 🤔

Reply:
Hmm, the rest of the reserved html characters gets interpreted as intended, although you're right that it might be just better to escape them altogether and never worry what browser is the user using and some misfortunate syntax in documented cairo code itself. Good call, I will adjust accordingly.